output/cloud/expv2/output.go

```diff
@@ -100,12 +117,49 @@ func (o *Output) Start() error {
 	o.runFlushWorkers()
 	o.periodicInvoke(o.config.AggregationPeriod.TimeDuration(), o.collectSamples)
 
+	if o.tracingEnabled() {
+		testRunID, err := strconv.ParseInt(o.referenceID, 10, 64)
+		if err != nil {
+			return err
+		}
+		o.requestMetadatasCollector = newRequestMetadatasCollector(testRunID)
+
+		insightsClientConfig := insights.ClientConfig{
+			IngesterHost: o.config.TracesHost.ValueOrZero(),
+			AuthConfig: insights.ClientAuthConfig{
+				Enabled:                  true,
+				TestRunID:                testRunID,
+				Token:                    o.config.Token.ValueOrZero(),
+				RequireTransportSecurity: true,
+			},
+			TLSConfig: insights.ClientTLSConfig{
+				Insecure: false,
+			},
+		}
+		insightsClient := insights.NewClient(insightsClientConfig)
+
+		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+		defer cancel()
+		if err := insightsClient.Dial(ctx); err != nil {
+			return err
+		}
+
+		o.shutdownHooks = append(o.shutdownHooks, func() error {
+			return insightsClient.Close()
+		})
```

Comment:
We are storing a state here, why not use the classic way of storing it in a field and then use the Stop function following the defined lifecycle from the Output interface?

Reply:
You're right. Simplified in [de2257a](https://github.com/grafana/k6/pull/3100/commits/de2257a6d0b07cfd2972a724b14354f9917d503c).